hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/crypto/key/kms/ValueQueue.java

```diff
@@ -269,12 +269,23 @@ public ValueQueue(final int numValues, final float lowWaterMark, long expiry,
    * Initializes the Value Queues for the provided keys by calling the
    * fill Method with "numInitValues" values
    * @param keyNames Array of key Names
-   * @throws ExecutionException executionException.
+   * @throws IOException if no successful initialization for any key
    */
-  public void initializeQueuesForKeys(String... keyNames)
-      throws ExecutionException {
+  public void initializeQueuesForKeys(String... keyNames) throws IOException {
+    int successfulInitializations = 0;
+    ExecutionException lastException = null;
+
     for (String keyName : keyNames) {
-      keyQueues.get(keyName);
+      try {
+        keyQueues.get(keyName);
+        successfulInitializations++;
+      } catch (ExecutionException e) {
+        lastException = e;
+      }
+    }
+
+    if (keyNames.length > 0 && successfulInitializations == 0) {
+      throw new IOException("Failed to initialize any queue for the provided keys.", lastException);
```

Comment:
It seems you've made warm up a best error operation. If so, there should be no need to through an exception here. Just logging a warning should be enough.

Reply:
So this is used to indicate all key initialization failed and should be retried. 
Since the return value is void of this func and it's caller funcs, keeping the IOException. Otherwise, it's hard to update the `success` flag in [.../hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSDirEncryptionZoneOp.java](https://github.com/apache/hadoop/pull/6774/files/10d763a5ff514541ef1eea11d70bf5173374a5d1#diff-092663652ffe33b10e51bfa062724d70e6334a9a14e64db35c9854805e09da14)